src/main/java/com/amuzil/omegasource/magus/radix/RadixPath.java

```diff
@@ -0,0 +1,75 @@
+package com.amuzil.omegasource.magus.radix;
+
+import com.amuzil.omegasource.magus.skill.forms.Form;
+import com.amuzil.omegasource.magus.skill.forms.FormDataRegistry;
+import com.amuzil.omegasource.magus.skill.modifiers.ModifiersRegistry;
+import com.amuzil.omegasource.magus.skill.modifiers.api.ModifierData;
+import com.ibm.icu.impl.Pair;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.ListTag;
+import net.minecraftforge.common.util.INBTSerializable;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+public class RadixPath implements INBTSerializable<CompoundTag> {
```

Comment:
Ideally, each path corresponds to a Skill, right? So where are we storing the Skill activator node/how or where are we activating the Skill

Reply:
The Node on the tree would activate the Skill, this RadixPath is a serializable dataset that allows us to recreate the steps taken through the tree to reach that same node with the same data in each preceding node